### PR TITLE
Command-line overrides and "install --copy" with rsync

### LINKED ISSUE #6: Feature request: instaclone rsync
Does everything that `instaclone install` does, but instead of symlinking, does an rsync (with deletes) on local node_modules.

This way I can use an instaclone as a starting point to do npm installs or upgrades and then do a new instaclone publish.

### PR DESCRIPTION
**Pull request by jlevy**
Fixes #6 and #7

### CODE REVIEW

**Comment by jlevy** [comment]
@marcello3d @mblakele

File: instaclone/configs.py [hunk 1]
```diff
@@ -13,19 +13,35 @@
...
-
-CONFIG_VERSION_RE = re.compile("^[\\w.-]+$")
+CONFIG_DEFAULTS = {
+  "install_method": "symlink"
+}
+CONFIG_DESCRIPTIONS = {
+  "local_path": "the local target path to sync to, relative to current dir",
+  "remote_path": "remote path (in backing store such as S3) to sync to",
+  "remote_prefix": "remote path prefix (such as s3://my-bucket/instaclone) to sync to",
+  "install_method": "the way to install files, either 'symlink' or 'copy'",
```
Comment (marcello3d) [inline instaclone/configs.py]: there are more options right?

File: instaclone/configs.py [hunk 2]
```diff
@@ -93,9 +112,17 @@ def _load_raw_configs(override_path):
   try:
     items = parsed_configs["items"]
     for config_dict in items:
-      # TODO: Could overlay global configs and local ones here.
-      combined = {key: None for key in Config._fields}
-      combined.update(config_dict)
+      # Legacy fix for renamed key. TODO: Remove this after a while.
+      if "copy_type" in config_dict:
+        config_dict["install_method"] = config_dict["copy_type"]
```
Comment (marcello3d) [inline instaclone/configs.py]: maybe log a deprecation warning?

File: instaclone/instaclone.py [hunk 3]
```diff
@@ -97,17 +97,29 @@ def _decompress_dir(archive_path, target_path, force=False):
     ARCHIVER.unarchive(archive_path, temp_dir)
 
 
+def _rsync_dir(source_dir, target_dir):
+  popenargs = ["rsync", "-a", "--delete", "--chmod=u+w"]
```
Comment (marcello3d) [inline instaclone/instaclone.py]: what is the chmod for? shouldn't the permissions match whatever the original bundle was?

**Comment by marcello3d** [comment]
Would be nice to see something in the readme about how to use the new feature. I'm a little confused about the difference between `copy`/`fastcopy` in config, and `--copy`

File: instaclone/instaclone.py (hunk 3 above)
Comment (mblakele) [inline instaclone/instaclone.py]: Looks to me like `_rsync_dir` copies from a cache dir (which is still read-only?):  https://github.com/vivlabs/instaclone/pull/9/files#diff-b1f4498e55edb22685ab044369d982f8R150

**Comment by mblakele** [comment]
Might be worth mentioning that you can testing this with `pip install git+https://github.com/vivlabs/instaclone.git@fee5e53`. I've done that on my laptop.

File: instaclone/configs.py (hunk 1 above)
Comment (jlevy) [inline instaclone/configs.py]: These are the config setting descriptions only. (If you run instaclone --help it should be clear.)

File: instaclone/instaclone.py (hunk 3 above)
Comment (jlevy) [inline instaclone/instaclone.py]: Yes, this is a fast way to make the files writable since they are read-only in cache. Will add a comment.

Comment (marcello3d) [inline instaclone/instaclone.py]: does it keep things +x?

Comment (mblakele) [inline instaclone/instaclone.py]: Seems to:

```
$ rm -rf node_modules
$ instaclone install --copy
using config file: ./instaclone.yml
using cache: /Users/mblakele/.instaclone/cache
using rsync for faster copy
installed from cache (fastcopy): node_modules -> /Users/mblakele/.instaclone/cache/contents/s3/sf-builds/instaclone/fubar/node_modules.$a6273aed294f656b5bd9151145de72c8c1f88ade-Darwin$/node_modules
$ ls -l node_modules/.bin/ | head -1
total 216
lrwxr-xr-x  1 mblakele  staff  19 Oct  6 09:54 _mocha -> ../mocha/bin/_mocha
$ ls -l node_modules/mocha/bin/_mocha
-rwxr-xr-x  1 mblakele  staff  11860 Apr  8 00:32 www/node/node_modules/mocha/bin/_mocha
```

**Pull request merged**